M2Mqtt/Messages/MqttMsgContext.cs

```diff
@@ -43,7 +43,7 @@ public class MqttMsgContext
         /// <summary>
         /// Timestamp in ticks (for retry)
         /// </summary>
-        public int Timestamp { get; set; }
+        public float Timestamp { get; set; }
```

Comment:
If you are changing the type here because of `TickCount64` then the correct type is `long`.